test/test.ts

```diff
@@ -21,14 +21,14 @@ modules.forEach((exports, name) => {
     describe(name, function() {
       exports.configurations.forEach((cfg: any) => {
         it(JSON.stringify(cfg.settings), () => {
-          expect(compile(exports.schema, cfg.settings)).to.be.equal(cfg.types)
+          expect(compile(exports.schema, name, cfg.settings)).to.be.equal(cfg.types)
```

Comment:
Since we now use the file name for stuff (like rendering relative paths, and as a fallback in case we don't have an ID or title on the schema) it has to be passed into `compile`.